pkg/reflector/dml_source.go

```diff
@@ -122,3 +127,37 @@ func (source *sqlDmlSource) Next(ctx context.Context) (statement schema.DMLState
 	err = errNoNewStatements
 	return
 }
+
+// Helper function to generate the SQL query
+func generateSQLQuery(ledgerTableName, shardingFamily, shardingTable string, blocksize int) string {
+	baseQuery := "SELECT seq, leader_ts, statement, family_name, table_name FROM $1 WHERE "
+	whereClause := "seq > ?"
+	limitClause := " ORDER BY seq LIMIT $4"
+
+	if shardingFamily != "" {
+		whereClause += fmt.Sprintf(" AND family_name IN $2")
+	}
+
+	if shardingTable != "" {
+		whereClause += fmt.Sprintf(" AND CONCAT(family_name,'___',table_name) IN $3")
```

Comment:
I have consulted Copilot and confirmed that using CONCAT will indeed slow down the query performance:
<img width="925" alt="Screenshot 2024-02-26 at 6 30 28 PM" src="https://github.com/segmentio/ctlstore/assets/6007569/dd05b5ab-79c0-4704-9753-d9da32baf840">

But I will keep it for now, leaving us two options to optimize:

1. As suggested by Copilot, add an additional column `family_table_concat` beside `family_name` and `table_name` to the ledger table.
2. Utilize composite key suggested by Erik.

Either way making necessary code changes doesn't feel like a too time-consuming task.

Reply:
Or we can change the `table` column in the ledger table to be the fully-qualified name.